test/ObserverSpec.ts

```diff
@@ -12,13 +16,44 @@ describe("@Observer", () => {
 
         expect(initial).to.be.equal(0);
         expect(result).to.be.equal(999);
-
-        ServiceStatus.instance.cancelInterval();
     });
 
     it("should set default value of sample observer as 0", () => {
         let observer = new SampleObserver();
         expect(observer.state).to.be.equal(0);
-        ServiceStatus.instance.cancelInterval();
     });
+});
+
+describe("@ObserverStatusForAllClasses", () => {
+    afterEach(async () => ServiceStatus.instance.cancelInterval());
+
+    it("should update all annotated classes when service status changes - Two observers from same class", () => {
+        let observerOne = new SampleObserver();
+        let observerTwo = new SampleObserver();
+        let initialStateOne = observerOne.state;
+        let initialStateTwo = observerTwo.state;
+
+        ServiceStatus.instance.goOnline();
+        let resultObserverOne = observerOne.state;
+        let resultObserverTwo = observerTwo.state;
+        ServiceStatus.instance.goOffline();
+
+        expect(initialStateOne).to.be.equal(0);
```

Comment:
we don't test that it is equal to zero.
let's set this value from the test itself.
we onl'y test the transition. i.e.
let o1 = new SampleObserver()
o1.state = 0;
let o2 = new SampleObserver()
o2.state = 10;